google/cloud/internal/timer_queue.cc

```diff
@@ -46,80 +56,83 @@ void TimerQueue::Shutdown() {
   std::unique_lock<std::mutex> const lk(mu_);
   shutdown_ = true;
   cv_.notify_all();
+  cv_follower_.notify_all();
 }
 
-// Timers added via Schedule should be managed by one or more threads that
-// call Service. Calls to Service only return after Shutdown has been called.
 void TimerQueue::Service() {
-  while (true) {
-    std::unique_lock<std::mutex> lk(mu_);
-    auto const until = NextExpiration(lk);
-    // TODO(#10521): Predicate condition ne < until may need to be reworked.
-    auto predicate = [this, until, &lk] {
-      auto const ne = NextExpiration(lk);
-      return shutdown_ || ne < until ||
-             HasExpiredTimer(lk, std::chrono::system_clock::now());
+  // It is easier to understand this if you first ignore shutdown and empty
+  // timer queues.
```

Comment:
I would suggest dropping this preamble.  We address shutdown below, and never mention empty queues again.

Reply:
Done.